test/testdata/infer/lub_tuple_array.rb

```diff
@@ -0,0 +1,27 @@
+# typed: true
+extend T::Sig
+
+sig {
+  type_parameters(:U)
+  .params(
+    x: T.any(
+      T.type_parameter(:U),
+      T::Array[T.type_parameter(:U)]
+    )
+  )
+  .returns(T::Array[T.type_parameter(:U)])
+}
+def foo(x)
+  res = case x
+  when Array
+    # This type is particularly unweildly, and I'd love to make it smaller if possible.
```

Comment:
```suggestion
    # This type is particularly unwieldy, and I'd love to make it smaller if possible.
```

Reply:
wow TIL; i always thought it was unweildly